app/components/modals/new-item/component.js

```diff
@@ -1,22 +1,50 @@
 import Ember from "ember";
+import UniqueFieldValidator from "last-strawberry/validators/unique-field-validator";
+import computed from "ember-computed-decorators";
 
 export default Ember.Component.extend({
+  session:     Ember.inject.service(),
+
   classNames: ["col"],
 
+  @computed("codeValidator.isValid", "changeset.isValid")
+  isValid(validCode, validChangeset) {
+    return validCode && validChangeset;
+  },
+
   didInsertElement() {
+    this._super(...arguments);
```

Comment:
I don't think we need to call super here. This is only needed from init overrides